docs/components/plugins/AppConfigPropsTable.js

```diff
@@ -0,0 +1,124 @@
+function appendProp(formattedSchema, property, _level) {
+  let level = _level;
+  const propertyKey = property[0];
+  const propertyValue = property[1];
+
+  //don't append deprecated or "hidden" properties
+  if (propertyValue.meta && (propertyValue.meta.deprecated || propertyValue.meta.hidden)) {
+    return;
+  }
+
+  //append passed-in property
+  formattedSchema.push({
+    //The ` quotes are for markdown highlighting, and #### is to apply anchor linking only on top-level properties
+    name: level ? `\`${propertyKey}\`` : `#### \`${propertyKey}\``,
+    type: propertyValue.enum ? 'enum' : propertyValue.type,
+    description: smartlyCreateDescription(property),
+    level: level,
+  });
+
+  //increase nesting level for sub-properties
+  level++;
+
+  //recursively apply appending logic for each sub-property (if any) -> Note: sub-props are sometimes nested within "items"
+  if (propertyValue.properties) {
+    Object.entries(propertyValue.properties).forEach(subproperty => {
+      appendProp(formattedSchema, subproperty, level);
+    });
+  } else if (propertyValue.items && propertyValue.items.properties) {
+    Object.entries(propertyValue.items.properties).forEach(subproperty => {
+      appendProp(formattedSchema, subproperty, level);
+    });
+  }
+}
+
+//setting up a property's formatted description value, with all the possible extra values
+function smartlyCreateDescription(property) {
+  const propertyValue = property[1];
+
+  var propertyDescription = propertyValue.description;
+  if (propertyValue.exampleString) {
+    propertyDescription += `\n\n >` + propertyValue.exampleString;
+  }
+  if (propertyValue.meta && propertyValue.meta.regexHuman) {
+    propertyDescription += `\n\n` + propertyValue.meta.regexHuman;
+  }
+  if (propertyValue.meta && propertyValue.meta.expoKit) {
+    propertyDescription += `\n` + `>**ExpoKit**: ` + propertyValue.meta.expoKit;
+  }
+  if (propertyValue.meta && propertyValue.meta.bareWorkflow) {
+    propertyDescription += `\n` + `>**Bare workflow**: ` + propertyValue.meta.bareWorkflow;
+  }
+
+  return propertyDescription;
+}
+
+export default class AppConfigPropsTable extends React.Component {
+  render() {
+    var rawSchema = Object.entries(this.props.schema);
+    var formattedSchema = [];
+
+    //appends each schema property (each index will become a tablerow)
+    rawSchema.map(property => {
+      appendProp(formattedSchema, property, 0);
+    });
+
+    return (
+      <div>
+        <table className={STYLES_TABLE}>
+          <thead>
+            <tr>
+              <td>Property</td>
+              <td>Type</td>
+              <td>Description</td>
+            </tr>
+          </thead>
+          <tbody>
+            {formattedSchema.map((property, index) => {
+              return (
+                <tr key={index}>
+                  <td>
+                    <div
+                      data-testid={property.name.match(/[a-zA-Z]+/g)}
```

Comment:
```suggestion
                      data-testid={property.name}
```

Why is `.match(/[a-zA-Z]+/g)` needed here? This value should be a string instead of an array

Reply:
Currently this is because the property.name might equal something like: \`#### androidNavigationBar\` OR \`visible\`, so in the `.test` file I wanted to just be able to read these as just "androidNavigationBar" or "visible" for the reader's sake, but I can find another way to be clear